python/pyspark/pandas/tests/test_dataframe.py

```diff
@@ -5779,14 +5779,183 @@ def test_astype(self):
+                ],
+                "B": [
+                    pd.Timestamp("2021-11-20"),
+                    pd.Timestamp("2023-06-02"),
+                    pd.Timestamp("2026-07-11"),
+                    pd.Timestamp("2026-07-11"),
+                ],
+            }
+        )
+        pdf = psdf.to_pandas()
+        # NOTE: Set `datetime_is_numeric=True` for pandas:
+        # FutureWarning: Treating datetime data as categorical rather than numeric in `.describe` is deprecated
+        # and will be removed in a future version of pandas. Specify `datetime_is_numeric=True` to silence this
+        # warning and adopt the future behavior now.
+        # NOTE: Compare the result except percentiles, since we use approximate percentile
+        # so the result is different from pandas.
+        self.assert_eq(
+            psdf.describe().loc[["count", "mean", "min", "max"]],
+            pdf.describe(datetime_is_numeric=True).astype(str).loc[["count", "mean", "min", "max"]],
+        )
+
+        # String & timestamp columns
+        psdf = ps.DataFrame(
+            {
+                "A": ["a", "b", "b", "c"],
+                "B": [
+                    pd.Timestamp("2021-11-20"),
+                    pd.Timestamp("2023-06-02"),
+                    pd.Timestamp("2026-07-11"),
+                    pd.Timestamp("2026-07-11"),
+                ],
+            }
+        )
+        pdf = psdf.to_pandas()
+        self.assert_eq(
+            psdf.describe().loc[["count", "mean", "min", "max"]],
+            pdf.describe(datetime_is_numeric=True).astype(str).loc[["count", "mean", "min", "max"]],
+        )
+        psdf.A += psdf.A
+        pdf.A += pdf.A
+        self.assert_eq(
+            psdf.describe().loc[["count", "mean", "min", "max"]],
+            pdf.describe(datetime_is_numeric=True).astype(str).loc[["count", "mean", "min", "max"]],
+        )
+
+        # Numeric & timestamp columns
+        psdf = ps.DataFrame(
+            {
+                "A": [1, 2, 2, 3],
+                "B": [
+                    pd.Timestamp("2021-11-20"),
+                    pd.Timestamp("2023-06-02"),
+                    pd.Timestamp("2026-07-11"),
+                    pd.Timestamp("2026-07-11"),
+                ],
+            }
+        )
+        pdf = psdf.to_pandas()
+        pandas_result = pdf.describe(datetime_is_numeric=True)
+        pandas_result.B = pandas_result.B.astype(str)
+        self.assert_eq(
+            psdf.describe().loc[["count", "mean", "min", "max"]],
+            pandas_result.loc[["count", "mean", "min", "max"]],
+        )
+        psdf.A += psdf.A
+        pdf.A += pdf.A
+        pandas_result = pdf.describe(datetime_is_numeric=True)
+        pandas_result.B = pandas_result.B.astype(str)
+        self.assert_eq(
+            psdf.describe().loc[["count", "mean", "min", "max"]],
+            pandas_result.loc[["count", "mean", "min", "max"]],
+        )
+
+        # Empty DataFrame
```

Comment:
@itholic last comment. Can we split this test into multiple test? this one unittest is too long, and would be difficult to debug when a test fails.

Reply:
Sure, refined!